smilekit/family_info/views.py

```diff
@@ -459,24 +454,28 @@ def dashboard(request):
     return HttpResponse(t.render(c))
 
 
-@login_required
-def end_interview(request, **args):
-    """Attempt to store the responses posted and, on success, redirect
-    to the list of families."""
-    rp = request.POST
-    answer_count = 0
-
+def get_visits(rp, request):
     if 'visit_id' in rp:
         try:
-            visits = [Visit.objects.get(pk=rp['visit_id'])]
+            return [Visit.objects.get(pk=rp['visit_id'])]
         except ValueError:  # for old visits where no id is stored:
-            visits = [
+            return [
                 v for v in request.user.visit_set.all() if v.is_happening]
         except Visit.DoesNotExist:
             # this can occasionally happen due to selenium test teardown.
-            visits = []
+            return []
     else:
-        visits = [v for v in request.user.visit_set.all() if v.is_happening]
+        return [v for v in request.user.visit_set.all() if v.is_happening]
```

Comment:
Could be re-written as `return request.user.visit_set.filter(is_happening=True)`